spec/api.json

```diff
@@ -491,20 +491,15 @@
         "description": "Indicates, that the UI flow could be continued by showing a recovery ui",
         "properties": {
           "action": {
-            "description": "Action will always be `show_recovery_ui`\nshow_recovery_ui ContinueWithActionShowRecoveryUIString",
-            "enum": [
-              "show_recovery_ui"
```

Comment:
The enums are gone!

Reply:
Correct. However, the generated code works in Go and Typescript:

```typescript
export type ContinueWith =
  | ({ action: "set_ory_session_token" } & ContinueWithSetOrySessionToken)
  | ({ action: "show_recovery_ui" } & ContinueWithRecoveryUi)
  | ({ action: "show_settings_ui" } & ContinueWithSettingsUi)
  | ({ action: "show_verification_ui" } & ContinueWithVerificationUi)

export function ContinueWithFromJSON(json: any): ContinueWith {
  return ContinueWithFromJSONTyped(json, false)
}

export function ContinueWithFromJSONTyped(
  json: any,
  ignoreDiscriminator: boolean,
): ContinueWith {
  if (json === undefined || json === null) {
    return json
  }
  switch (json["action"]) {
    case "set_ory_session_token":
      return {
        ...ContinueWithSetOrySessionTokenFromJSONTyped(json, true),
        action: "set_ory_session_token",
      }
    case "show_recovery_ui":
      return {
        ...ContinueWithRecoveryUiFromJSONTyped(json, true),
        action: "show_recovery_ui",
      }
    case "show_settings_ui":
      return {
        ...ContinueWithSettingsUiFromJSONTyped(json, true),
        action: "show_settings_ui",
      }
    case "show_verification_ui":
      return {
        ...ContinueWithVerificationUiFromJSONTyped(json, true),
        action: "show_verification_ui",
      }
    default:
      throw new Error(
        `No variant of ContinueWith exists with 'action=${json["action"]}'`,
      )
  }
}
```

Usage in Go:
```go
	flow, res, err := client.FrontendApi.CreateNativeRecoveryFlow(ctx).Execute()
	pkg.SDKExitOnError(err, res)

	for _, continueWith := range flow.ContinueWith {
		switch action := continueWith.GetActualInstance().(type) {
		case ory.ContinueWithSetOrySessionToken:
			fmt.Println("ContinueWithSetOrySessionToken")
			sessionToken := action.GetOrySessionToken()
			_ = sessionToken
		case ory.ContinueWithSettingsUi:
			fmt.Println("ContinueWithSettingsUi")
			pkg.PrintJSONPretty(action.GetFlow())

		default:
			fmt.Println("default")
			pkg.PrintJSONPretty(action)
		}
	}
```

I coulnd't figure out a "better" way to make this work otherwise. This is also how the `uiNodeAttributes` have worked since forever.

Reply:
actually I might have to twiddle with this some more... fuck

Reply:
OK I think I got it. Golang now also correctly uses the `action` field to decide among the elements of the type union.

Reply:
OK - this is a breaking change then right? If so, we'll need to document it in some sense so that we can publish an upgrade advisory for Ory Network

Reply:
@alnr what exactly is the point of removing these enums? I could not really figure it out.

Reply:
I was under the impression that the generated code in Go would be invalid without changing the enums, but that may be incorrect.